lib/galaxy/config.py

```diff
@@ -276,6 +276,11 @@ def __init__( self, **kwargs ):
         # On can mildly speed up Galaxy startup time by disabling index of help,
         # not needed on production systems but useful if running many functional tests.
         self.index_tool_help = string_as_bool( kwargs.get( "index_tool_help", True ) )
+        self.tool_name_boost = kwargs.get( "tool_name_boost", None )
```

Comment:
This is exactly how I would do it, to me this seems like the best design.

I routinely design components that work even if ``config`` does not have configuration values defined  (e.g. using ``getattr`` instead of attribute accession directly) - I do this so the components work in a broader situations and the dependency between the component and Galaxy's ``config`` god object is reduced. In such cases the component itself must define defaults. Then the decision is whether to duplicate the defaults in ``config.py`` or just set them to ``None`` and let the component decide. They already unfortunately have to appear twice (we don't auto-generate ``config.ini.sample`` comments) - I don't see any value in compounding this problem by re-duplicating the logic yet again.